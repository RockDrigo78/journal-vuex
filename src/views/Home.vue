<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png" />
    <h1>Hola Ro</h1>
    <button class="btn btn-primary" @click="goToDaybook">Primary</button>
    <button class="btn btn-secondary" @click="goToDaybook">Secondary</button>
    <button class="btn btn-success" @click="goToDaybook">Daybook</button>
  </div>
</template>

<script>
export default {
  name: "Home",
  components: {},
  methods: {
    goToDaybook() {
      this.$router.push({ name: "daybook" });
    },
  },
};
</script>
