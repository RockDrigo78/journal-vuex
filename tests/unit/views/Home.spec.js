import { shallowMount } from '@vue/test-utils';
import Home from '@/views/Home.vue';

describe('Home component tests', () => {
    test('should match snapshot', () => {
        const wrapper = shallowMount(Home);
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('should redirect to no-entry route when button clicked', () => {
        const mockRouter = {
            push: jest.fn(),
        };

        const wrapper = shallowMount(Home, {
            global: {
                mocks: {
                    $router: mockRouter,
                },
            },
        });

        wrapper.find('button').trigger('click');

        expect(mockRouter.push).toHaveBeenCalled();
        expect(mockRouter.push).toHaveBeenCalledWith({ name: 'daybook' });
    });
});
