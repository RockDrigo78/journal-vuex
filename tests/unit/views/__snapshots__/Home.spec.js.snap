// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component tests should match snapshot 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1>Hola Ro</h1><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">Secondary</button><button class="btn btn-success">Daybook</button>
</div>
`;
